projectile.el

```diff
@@ -4529,6 +4529,7 @@ See `projectile--cleanup-known-projects'."
   "Clear both `projectile-known-projects' and `projectile-known-projects-file'."
   (interactive)
   (setq projectile-known-projects nil)
+  (projectile-discover-projects-in-search-path)
```

Comment:
Perhaps you can add a prefix argument to the command, which would trigger the project discovery.

Reply:
Should `(projectile-save-known-projects)` execute in both paths?

Reply:
I don't remember if the discover function saves what it finds or not. Depends on this.

Reply:
If `(projectile-discover-projects-in-search-path)` finds a project to add it does save.

Reply:
Should neither `(projectile-save-known-projects)` or `(projectile-discover-projects-in-search-path)` execute when no prefix argument is provided?